GitHub actions

MartinNowack opened the pull request:
Adds support for GitHub Actions - up to now, everything builds.

In addition it fixed a coupled of issues:
- For `LLVM_VERSION` it's sufficient to contain only the major version
- Builds on macOS are more robust by detecting the correct SDK used
- Use non-interactive version for Docker build

Thanks for @jordr original effort I could build on.

<!--Thank you for contributing to KLEE.  We are looking forward to reviewing your PR.  However, given the small number of active reviewers and our limited time, it might take a while to do so.  We aim to get back to each PR within one month, and often do so within one week. 

To help expedite the review please ensure the following, by adding an "x" for each completed item:
-->
- [x] The PR addresses a single issue. <!-- In other words, if some parts of a PR could form another independent PR, you should break this PR into multiple smaller PRs. -->
- [x] There are no unnecessary commits. <!-- For instance, commits that fix issues with a previous commit in this PR are unnecessary and should be removed (you can find [documentation on squashing commits here](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request#squash-your-changes)). -->
- [x] Larger PRs are divided into a logical sequence of commits.
- [x] Each commit has a meaningful message documenting what it does.
- [x] The code is commented.   <!-- In particular, newly added classes and functions should be documented. -->
- [ ] The patch is formatted  <!--  via  [clang-format](https://clang.llvm.org/docs/ClangFormat.html) (see also [git-clang-format](https://raw.githubusercontent.com/llvm/llvm-project/master/clang/tools/clang-format/git-clang-format) for Git integration).  Please only format the patch itself and code surrounding the patch, not entire files.  Divergences from clang-formatting are only rarely accepted, and only if they clearly improve code readability. -->
- [ ] Add test cases exercising the code you added or modified.  <!--  We expect [system and/or unit test cases](https://klee.github.io/docs/developers-guide/#regression-testing-framework) for all non-trivial changes.  After you submit your PR, you will be able to see a [Codecov report](https://docs.codecov.io/docs/pull-request-comments) telling you which parts of your patch are not covered by the regression test suite.  You will also be able to see if the Travis CI and Cirrus CI tests have passed.  If they don't, you should examine the failures and address them before the PR can be reviewed.  -->
- [x] Spellcheck all messages added to the codebase, all comments, as well as commit messages.

jordr commented:
Hey! We should probably update the readme as well to include Github Actions badges instead of Travis.

ccadar commented:
@MartinNowack @jordr what needs to happen for this to appear as a check?

kren1 commented on .github/workflows/build.yaml:
```diff
@@ -0,0 +1,191 @@
+name: Continues Integration
```
Continuous

MartinNowack replied:
@kren1 Thanks! Fixed ;)

MartinNowack commented:
@ccadar As soon as the CI passes - it can be merged.

I added an entry to `README.md`. As soon as this PR is merged and the CI passes, it should appear.

And as soon as this PR is merged, I'll open an additional PR to remove Travis and do some clean-up as needed.

ccadar commented:
Great, it seems to be working!  Thanks, @MartinNowack and @jordr !
I will disable the Travis CI check and make these mandatory -- however, perhaps we shouldn't remove the Travis CI files right away, let's wait for a couple of months to make sure everything works fine with GH Actions.

MartinNowack commented:
@ccadar - I added some additional updates and removed several restrictions we had with Travis CI.
The current timeout for Github Actions is 6h. The overall build time is much faster. Beside that, quite often many more jobs run in parallel - 21 jobs instead of 5 on Travis CI.

This PR is ready.

ccadar commented:
@MartinNowack this looks awesome!  It seems faster, with less timeout restrictions and with a nicer interface than before!
So let's merge it and get back to work! ;)

ccadar commented:
@MartinNowack one other question: what happened to CodeCov, I see it didn't produce a report -- was it tied in any way to Travis?

MartinNowack commented:
@ccadar There is still something off with the codecov - I already noticed it last night. It might be an authentication issue. The error message is that it cannot detect the CI environment. Although it is supported.

Still, it's only the uploading - generating the coverage information works as expected.

MartinNowack commented:
Forgot to mention explicitly, Github Actions is supported (including via Docker) - there is just something off.
